graylog2-server/src/main/java/org/graylog2/streams/filters/StreamDestinationFilterService.java

```diff
@@ -175,4 +183,13 @@ public void forEachEnabledFilterGroupedByStream(Consumer<GroupByStreamResult> co
                 ))
         ), GroupByStreamResult.class).forEach(consumer);
     }
+
+    @Subscribe
+    @SuppressWarnings("unused")
+    public void handleStreamDeleted(StreamDeletedEvent streamDeletedEvent) {
```

Comment:
Can we directly just deleted it? That shouldn't be coming from an api call, so I don't think there is any need to fetch the filter, verify if the filter matches the streamId before deleting it 
````
    public void handleStreamDeleted(StreamDeletedEvent streamDeletedEvent) {
        collection.deleteOne(Filters.eq(FIELD_STREAM_ID, streamDeletedEvent.streamId()));

    }
    
    ````

Reply:
Good point. Except, are we sure there is only a single rule per stream? I think we should call `deleteMany`.

Reply:
oh right, you are absolutely right, we should delete all rules